pkg/cpu_collector/cpu_collector.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2021 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/pingcap/log"
+	"github.com/shirou/gopsutil/v3/cpu"
+	"github.com/tikv/pd/pkg/errs"
+	"go.uber.org/zap"
+)
+
+// CPUCollector is a daemon to collect the CPU usage of a PD server within a certain interval.
+type CPUCollector struct {
+	// Make sure every collector only has one background goroutine to keep collecting and updating the usage.
+	once     sync.Once
```

Comment:
Could you give an example about there are more than one goroutine?

Reply:
Considering that the packages under pkg are somewhat generic, so the main purpose is to prevent the caller from starting the same Collector multiple times. This is not currently a possibility within PD.

Reply:
Perhaps, it should be guaranteed by the caller?